lib/winston/logger.js

```diff
@@ -642,7 +627,9 @@ class Logger extends Transform {
 
   _addDefaultMeta(msg) {
     if (this.defaultMeta) {
-      Object.assign(msg, this.defaultMeta);
+      // The msg must be cloned as it is being mutated, but any metadata provided with the msg takes precedence over default
+      const msgClone = Object.assign({}, msg);
```

Comment:
This extra call to Object.assign doesn't make sense to me, even with the comment. 
The next line is also using Object.assign, so you get exactly the same extent of property-copying including the same limits (e.g. that if `msg` contains object references, only the reference to the object is copied rather than doing a deep copy; mutations referenced by the comment would still seem to be passed through.)

Reply:
I think these are fair criticisms and likely due to an incomplete understanding of the copying of object references. The intent here iirc was to do a deep copy but I'm still unaware of a best practice of doing it in an easy manner without traversing the object structure manually. Open to all sorts of suggestions here.

Reply:
Did some further more involved testing of this over break; it's an insufficient solution as you likely knew since it still persists the object references even though it is an improvement of the previous solution. Working through updating the solution to account for my new test cases.